### PR TITLE
Sd1240 plugin doc js

### PR DESCRIPTION
**Pull request by jstucke**
- docs for helper function modules "debug" and "mongo_config_parser"
- moved `database_structure` to `web_interface/`
- moved duplicate code from web_interface components to own module

### CODE REVIEW

File: src/web_interface/database_structure.py [hunk 1]
```diff
@@ -32,14 +33,14 @@ def _create_tree_structure(list_of_dot_separated_strings):
     return structure_tree
 
 
-def _attach_field_to_tree(field, subtree):
-    splitted_field = field.split('.', 1)
-    if len(splitted_field) == 1:
+def _attach_field_to_tree(field: str, subtree):
```
Comment (dorpvom) [inline src/web_interface/database_structure.py:36]: Why not type the subtree? It seems to be a (default)dict

**Review by dorpvom — APPROVED** [review]

File: src/helperFunctions/mongo_config_parser.py [hunk 2]
```diff
@@ -1,13 +1,26 @@
+from typing import Union
+
 import yaml
 
 
-def parse_yaml(file_path):
+def parse_yaml(file_path: str) -> Union[dict, list, None]:
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/helperFunctions/mongo_config_parser.py:6]: YAGNI -> 1.

File: src/helperFunctions/debug.py [hunk 3]
```diff
@@ -2,14 +2,15 @@
 import sys
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/helperFunctions/debug.py:2]: I agree -> moved to `logging.py`

File: src/web_interface/database_structure.py (hunk 1 above)
Comment (jstucke) [reply src/web_interface/database_structure.py:36]: added some type hints

**Pull request merged**